cumulus/ansible/tasks/playbooks/volumes/ec2/attach.yml

```diff
@@ -19,12 +18,48 @@
     - block:
         # make local action
         - ec2_vol:
+            name: "ec2_{{ profile_id }}_{{ volume_name }}"
+            volume_size: "{{ volume_size }}"
+            region: "{{ region }}"
+            zone: "{{ volume_zone }}"
+            instance: "{{ instance_id }}"
+            device_name: /dev/xvdf
+          delegate_to: localhost
+          register: vol
+          when: not volume_id
+
+        # If we already have a volume_id we can't pass the volume_size
+        - ec2_vol:
             id: "{{ volume_id }}"
+            name: "ec2_{{ profile_id }}_{{ volume_name }}"
             region: "{{ region }}"
+            zone: "{{ volume_zone }}"
             instance: "{{ instance_id }}"
             device_name: /dev/xvdf
           delegate_to: localhost
           register: vol
+          when: volume_id
+
```

Comment:
@cjh1 ansible has a syntax that might be useful here:  ```"{{ volume_id | default(omit) }}"```   This will exclude the ```id``` attribute from the module completely if the ```volume_id``` variable isn't set.  Should be able to use that with ```id``` and ```volume_size``` to combine these two modules.  Though if they're mutually exclusive i'd recommend putting in a check with a ```fail``` because ec2_vol's behavior might be undefined if both are specified.